### PR TITLE
doc: add first development guide

### PR DESCRIPTION
**Pull request by tixxdz**
Tetragon Open Source project was just released and it does not include
default contributing guidelines.

This patch fixes this by adding:

1. CONTRIBUTING.md file in the root directory as suggested by github documentation:
https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

2. Development guide under docs directory with a section on how to submit pull requests.

3. Moves the DEVELOP.md file from root directory to the `docs/contributing/development/` one.

Fixes: https://github.com/cilium/tetragon/issues/33

Signed-off-by: Djalal Harouni <djalal@cilium.io>

### CODE REVIEW

File: docs/contributing/development/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+make sure to read and apply the 
+[Developer’s Certificate of
+Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#developer-s-certificate-of-origin).
+
+
+## Development Setup
+
+Documentation for Development setup is [here][develop].
+
+[develop]: ./DEVELOP.md
```
Comment (willfindlay) [inline docs/contributing/development/README.md:50]: WDYT about folding DEVELOP.md into here as a flat section? I'd rather not have it buried behind two clicks.

**Review by willfindlay — APPROVED** [review]

**Comment by michi-covalent** [comment]
regarding the file location: https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors

> To help your project contributors do good work, you can add a file with contribution guidelines to your project repository's root, docs, or .github folder. 

i vote for putting CONTRIBUTING.md at the root directory to be consistent with cilium https://github.com/cilium/cilium/blob/master/CONTRIBUTING.md

**Comment by willfindlay** [comment]
> i vote for putting CONTRIBUTING.md at the root directory to be consistent with cilium https://github.com/cilium/cilium/blob/master/CONTRIBUTING.md

+1, this is also consistent with how many projects do it.

File: docs/contributing/development/README.md (hunk 1 above)
Comment (tixxdz) [reply docs/contributing/development/README.md]: @willfindlay sure, done.

**Comment by tixxdz** [comment]
@michi-covalent @willfindlay done, hope I did get it all.

**Review by michi-covalent — APPROVED** [review]

**Pull request merged**